### PR TITLE
fix: default strategy no longer uses out dated data

### PR DESCRIPTION
**Pull request by StephenCoady**
breaking: interfaces updated to accept a more complete data. now accepts an object.
breaking: getUpdateFunction updated to make certain vars optional. now accepts an object.

### Description

addresses https://github.com/aerogear/offix/issues/116 and https://github.com/aerogear/offix/issues/111

there are breaking changes in this PR.

updates to integration tests, fyi @jhellar

also @wtrocki @darahayes I have added yarn to the scripts - do we still want to do that? I am +100. I haven't added it as a dev dependency, let me know if you want me to.

EDIT: just saw on a separate PR you want to just go with npm but alias it locally. Shout here if you want me to change yarn -> npm on the files here.

##### Checklist

- [ ] `npm test` passes
- [ ] `npm run build` works
- [ ] tests are included
- [ ] documentation is changed or added

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
   "scripts": {
     "test": "nyc lerna run test",
     "coverage": "nyc report --reporter=text-lcov | coveralls",
-    "cleanInstall": "lerna exec npm install --ignore-scripts",
+    "cleanInstall": "lerna exec yarn install --ignore-scripts",
```
Comment (wtrocki) [inline package.json]: ❤️ . 

@darahayes Any voice against it?

**Comment by wtrocki** [comment]
Yarn link was so convinient. Tried and tested. With this change every subsequent offline edit ends with conflict. I'm not sure we should be documenting this as this is going to get addressed soon anyway.

File: packages/offix-offline/src/conflicts/strategies/ConflictResolutionStrategy.ts [hunk 2]
```diff
@@ -1,21 +1,14 @@
...
-   * @param base - base data before any client changes were applied
-   * @param server - the latest server data returned to the client
-   * @param client - the data the client tried to send to the server
-   * @param operation - [optional] the name of the operation you wish to perform the strategy on.
    */
-  resolve: (base: ConflictResolutionData,
-            server: ConflictResolutionData,
-            client: ConflictResolutionData,
-            operation?: string) => ConflictResolutionData;
+  resolve: (data: ConflictDataSet) => ConflictResolutionData;
```
Comment (wtrocki) [inline packages/offix-offline/src/conflicts/strategies/ConflictResolutionStrategy.ts]: Set is usually considered and array like object. How about just `ConflictData`?

Comment (wtrocki) [reply packages/offix-offline/src/conflicts/strategies/ConflictResolutionStrategy.ts]: No strong opinion :P

File: package.json (hunk 1 above)
Comment (darahayes) [reply package.json]: Coming to this late but go for it :)

**Pull request merged**